### PR TITLE
Remove versioned WP-Parse.ly (part of ext now) and add unversioned as a submodule

### PR DESCRIPTION
**Pull request by rinatkhaziev**
## Description

Now that we moved Parsely to the [ext](https://github.com/Automattic/vip-go-mu-plugins-ext) repo we should clean it up from this one.

This PR removes all versioned copies and adds a submodule as a fallback.

This may break tests and also we need to modify the loader to include the fallback to minimize potential breakage on customer CI side (integration tests and workflows).

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples.

### CODE REVIEW

**Review by rebeccahum — COMMENTED** [review]
Do we need any kind of version gating like we do in JP? https://github.com/Automattic/vip-go-mu-plugins/blob/ba26d4502b8f658526af3f7c6634bcb86bd74dea/jetpack.php#L20-L34

**Comment by rinatkhaziev** [comment]
@rebeccahum I'm not sure, I don't remember an instance of WP-Parsely bumping the WP version requirements but I think @acicovic would be better equipped to answer that.

**Comment by acicovic** [comment]
Since wp-parsely 3.0 we [require WP version 5.0](https://github.com/Parsely/wp-parsely/blob/3.0.0/wp-parsely.php) and this has not changed. The wp-parsely versions we have here are from 3.1 to 3.5.

**Review by rebeccahum — DISMISSED** [review]

**Review by rebeccahum — DISMISSED** [review]
@rinatkhaziev This doesn't work because it's expecting a versioned folder: https://github.com/Automattic/vip-go-mu-plugins/blob/da2ceb9444f0a1963a525da260b133d6d140623f/wp-parsely.php#L195

We'll have to modify the logic to account for it being `wp-parsely` or rename the submodule.

**Comment by rebeccahum** [comment]
Testing steps:
1) Add `add_filter( 'wpvip_parsely_load_mu', '__return_true' );` to `plugin-loader.php`
2) Check out PR and do not include `vip-go-mu-plugins-ext` in mu-plugins
3) Go to wp-admin and expect to see notice `The Parse.ly plugin is not active. You need to [provide your Parse.ly Dash Site ID](http://es-site.vipdev.lndo.site/wp-admin/options-general.php?page=parsely) before things get cooking.` that Parsely plugin has been loaded

**Comment by rinatkhaziev** [comment]
@acicovic can you take a look at the modified loader and think of any scenario this may break?

File: wp-parsely.php [hunk 1]
```diff
@@ -191,30 +191,35 @@ function maybe_load_plugin() {
...
+		$integration_type = Parsely_Loader_Info::INTEGRATION_TYPE_MUPLUGINS_SILENT;
+	}
 
-		// Require VIP's customizations over wp-parsely.
-		$vip_parsely_plugin = __DIR__ . '/vip-parsely/vip-parsely.php';
-		if ( is_readable( $vip_parsely_plugin ) ) {
-			require_once $vip_parsely_plugin;
-		}
+	// Require the actual wp-parsely plugin.
+	require_once $entry_file;
```
Comment (rinatkhaziev) [inline wp-parsely.php:217]: Does this need a file_exists check just to be extra safe?

Comment (rebeccahum) [reply wp-parsely.php:217]: @rinatkhaziev IMHO, that seems to be excessive given the `is_readable()` check up above.

**Comment by acicovic** [comment]
What happens if we're pinned to a specific wp-parsely version, using the `wpvip_parsely_version` filter?

```php
add_filter(
    'wpvip_parsely_version',
    function() {
        return '3.1';
    } 
);
```
Users use this to pin to old releases or test new ones.
Tip: If you go to Settings->Parse.ly, in the WP dashboard, the active plugin version is displayed on the top left of the page.

File: wp-parsely.php (hunk 1 above)
Comment (rinatkhaziev) [reply wp-parsely.php:217]: but we're only calling `is_readable`for the list of versioned files, not the fallback `$entry_file`

Comment (rebeccahum) [reply wp-parsely.php:217]: Ah, good catch.

File: wp-parsely.php [hunk 2]
```diff
@@ -211,7 +211,9 @@ function maybe_load_plugin() {
 	}
 
 	// Require the actual wp-parsely plugin.
-	require_once $entry_file;
+	if ( is_readable( $entry_file ) ) {
+		require_once $entry_file;
+	}
```
Comment (rinatkhaziev) [inline wp-parsely.php:216]: Maybe flip around? If not readable then return early, otherwise load and let the active versions and whatnot

**Comment by rebeccahum** [comment]
@acicovic It will still fall back to the default version if it cannot find that set version. But the pinning won't be successful (given the fact the version won't exist)

**Comment by rinatkhaziev** [comment]
LGTM but since I created the PR I can't approve it :) @Automattic/vip-platform-cantina

**Review by acicovic — APPROVED** [review]

**Comment by acicovic** [comment]
I approved but apparently this won't work as we need someone from the Cantina team :)

**Review by rebeccahum — APPROVED** [review]

**Pull request merged**